src/Materials/Node/nodeMaterial.ts

```diff
@@ -1606,13 +1606,52 @@ export class NodeMaterial extends PushMaterial {
      * Clear the current graph and load a new one from a serialization object
      * @param source defines the JSON representation of the material
      * @param rootUrl defines the root URL to use to load textures and relative dependencies
+     * @param hasFrames defines whether or not source is Frame or standard NodeMaterial
      */
-    public loadFromSerialization(source: any, rootUrl: string = "") {
+    public loadFromSerialization(source: any, rootUrl: string = "", hasFrames: boolean = false) {

```

Comment:
This is where I'm a bit stressed.
THe NodeMaterial should NOT know about frames

All the editor and UI should not live here..

The editorData field was meant to store data that the NodeEDitor should not care about.